cli/install/install.go

```diff
@@ -0,0 +1,800 @@
+			}
+		}
+		fmt.Println("Usage: tt install -f if you already have those packages installed")
+		return false
+	}
+	return true
+}
+
+// checkExisting checks if program is already installed in binary directory.
+func checkExisting(version string, dst string) bool {
+	if _, err := os.Stat(filepath.Join(dst, version)); os.IsNotExist(err) {
+		return false
+	} else {
+		return true
+	}
+}
+
+// ExecuteCommand executes program with given args in verbose or quiet mode.
+func ExecuteCommand(program string, isVerbose bool, logFile *os.File, workDir string,
+	args ...string) error {
+	cmd := exec.Command(program, args...)
+	if isVerbose {
+		cmd.Stdout = os.Stdout
+		cmd.Stderr = os.Stderr
+	} else {
+		cmd.Stdout = logFile
+		cmd.Stderr = logFile
+	}
+	if workDir == "" {
+		workDir, _ = os.Getwd()
+	}
+	cmd.Dir = workDir
+	err := cmd.Start()
+	if err != nil {
+		return err
+	}
+	err = cmd.Wait()
+	return err
+}
+
+// downloadRepo downloads git repository.
+func downloadRepo(repoLink string, tag string, dst string,
+	logFile *os.File, verbose bool) error {
+	var err error
+	if tag == "master" {
+		err = util.ExecuteCommand("git", verbose, logFile, dst, "clone",
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	} else {
+		err = util.ExecuteCommand("git", verbose, logFile, dst, "clone", "-b", tag,
+			"--depth=1", "-j", "18", repoLink,
+			"--recursive", dst)
+	}
+
+	return err
+}
+
+// copyBuildedTT copies tt binary.
+func copyBuildedTT(dst, path, version string, flags InstallationFlag,
+	logFile *os.File) error {
+	var err error
+	if _, err := os.Stat(dst); os.IsNotExist(err) {
+		err = os.Mkdir(dst, FolderRights)
+		if err != nil {
+			return fmt.Errorf("Unable to create bindir: %s", err)
+		}
+	}
+	if flags.Reinstall {
+		err = os.Remove(filepath.Join(dst, version))
+		if err != nil {
+			printLog(logFile.Name())
+			return err
+		}
+	}
+	err = util.CopyFilePreverse(filepath.Join(path, "tt"), filepath.Join(dst, version))
+	return err
+}
+
+// installTt installs selected version of tt.
+func installTt(version string, binDir string, flags InstallationFlag, distfiles string) error {
+
+	// Check binary directory.
+	if binDir == "" {
+		return fmt.Errorf("Install path parameter is empty.")
+	}
+	logFile, err := ioutil.TempFile("", "tarantool_install")
+	if err != nil {
+		return err
+	}
+	defer os.Remove(logFile.Name())
+
+	// Check if that version is already installed.
+	log.Infof("Checking existing...")
+	if checkExisting(version, binDir) && !flags.Reinstall {
+		log.Infof("%s version of tt already exists, updating symlink...", version)
+		err := util.CreateSymLink(version, binDir, "tt", true)
+		log.Infof("Done")
+		return err
+	}
+
+	// Check tt dependencies.
+	if !flags.Force {
+		log.Infof("Checking dependencies...")
+		if !programDependenciesInstalled("tt") {
+			return nil
+		}
+	}
+
+	path, err := os.MkdirTemp("", "tt_install")
+	if err != nil {
+		return err
+	}
+	os.Chmod(path, FolderRights)
```

Comment:
Please add a comment about what is wrong with the default permissions of temporary dir.

Reply:
added some comments for constant